### PR TITLE
Side-by-side installation support for EXE

### PR DESCRIPTION
**Pull request by nguyen-binh-minh**
Addressing MAGN-6578 and MAGN-6623:
Changes:
- Remove Vasari from component menu
- Replace hardcoded version number
- Update uninstall behavior
  (i.e. Dynamo 0.8.3 would uninstall 0.8.2, 0.8.1 or 0.8.0 before installation but not 0.7.x. Besides, 0.8.3 would not uninstall Dynamo version 0.8.3 and above)
- Move checking for older version of Dynamo to Setup Initialization step to provide quicker feedback to user

@sharadkjaiswal @Benglin PTAL
FYI @ikeough

### CODE REVIEW

File: tools/install/DynamoInstaller.iss [hunk 1]
```diff
@@ -89,17 +101,29 @@ Source: Extra\DynamoInstaller.ico; DestDir: {app}; Flags: ignoreversion overwrit
...
-Source: temp\samples\*.*; DestDir: {commonappdata}\Dynamo\0.8\samples; Flags: ignoreversion overwritereadonly recursesubdirs; Components: DynamoTrainingFiles
+Source: temp\samples\*.*; DestDir: {commonappdata}\Dynamo\{#Major}.{#Minor}\samples; Flags: ignoreversion overwritereadonly recursesubdirs; Components: DynamoTrainingFiles
 
 ;Other Custom Nodes
-Source: temp\definitions\*; DestDir: {commonappdata}\Dynamo\0.8\definitions; Flags: ignoreversion overwritereadonly recursesubdirs; Components: DynamoCore
+Source: temp\definitions\*; DestDir: {commonappdata}\Dynamo\{#Major}.{#Minor}\definitions; Flags: ignoreversion overwritereadonly recursesubdirs; Components: DynamoCore
+
+[Registry]
+Root: HKCU; Subkey: "Software\{#ProductName}\{#Major}.{#Minor}"; Flags: uninsdeletekey
+Root: HKCU; Subkey: "Software\{#ProductName}\{#Major}.{#Minor}"; ValueType: dword; ValueName: "installed"; ValueData: "1"
```
Comment (sharadkjaiswal) [inline tools/install/DynamoInstaller.iss]: Why do we need registry entry for value "installed"??

Comment (nguyen-binh-minh) [inline tools/install/DynamoInstaller.iss]: @sharadkjaiswal exe doesn't need this value but msi does, so to make the two installer similar we should have this value for the exe installer as well

**Comment by nguyen-binh-minh** [comment]
@sharadkjaiswal updated as required

File: tools/install/DynamoInstaller.iss [hunk 2]
```diff
@@ -187,66 +185,36 @@ begin
...
-  end else
-    Result := 1;
+  sUnInstPath := ExpandConstant('Software\Microsoft\Windows\CurrentVersion\Uninstall\{#ProductName} {#Major}.{#Minor}');
+  sUninstallString := '';
+  RegQueryStringValue(HKLM, sUnInstPath, 'UnInstallString', sUninstallString);
+    if (sUninstallString <> '') then
+	begin
+		if not RegQueryDWordValue(HKLM, sUnInstPath, 'RevVersion', revision) then
+			begin
+				MsgBox('The Dynamo version is corrupted?!. Aborting...', mbInformation, MB_OK);
```
Comment (sharadkjaiswal) [inline tools/install/DynamoInstaller.iss]: Couldn't determine the revision number for already installed {#ProductName} {#Major}.{#Minor} In order to proceed with the installation, you need to uninstall {#ProductName} {#Major}.{#Minor} manually.

File: tools/install/DynamoInstaller.iss [hunk 3]
```diff
@@ -187,66 +185,36 @@ begin
...
+				result := false
+			end;
+		if (revision >= {#Rev}) then
+			begin
+				sMsg := ExpandConstant('{#ProductName} {#Major}.{#Minor}.') + IntToStr(revision) + ' is already installed. Aborting'
```
Comment (sharadkjaiswal) [inline tools/install/DynamoInstaller.iss]: {#VersionInfoTextVersion} or higher is already installed. In order to proceed with the installation, you need to uninstall {#ProductName} {#Major}.{#Minor} manually.

File: tools/install/DynamoInstaller.iss [hunk 4]
```diff
@@ -187,66 +185,36 @@ begin
...
-  end else
-    Result := 1;
+  sUnInstPath := ExpandConstant('Software\Microsoft\Windows\CurrentVersion\Uninstall\{#ProductName} {#Major}.{#Minor}');
+  sUninstallString := '';
+  RegQueryStringValue(HKLM, sUnInstPath, 'UnInstallString', sUninstallString);
+    if (sUninstallString <> '') then
+	begin
+		if not RegQueryDWordValue(HKLM, sUnInstPath, 'RevVersion', revision) then
+			begin
+				MsgBox('The installed Dynamo version is corrupted. Aborting...', mbInformation, MB_OK);
```
Comment (sharadkjaiswal) [inline tools/install/DynamoInstaller.iss]: Couldn't determine the revision number for already installed {#ProductName} {#Major}.{#Minor} In order to proceed with the installation, you need to uninstall {#ProductName} {#Major}.{#Minor} manually.

File: tools/install/DynamoInstaller.iss [hunk 5]
```diff
@@ -187,66 +185,36 @@ begin
...
+				result := false
+			end;
+		if (revision >= {#Rev}) then
+			begin
+				sMsg := ExpandConstant('{#ProductName} version {#ProductVersion} or higher is already installed.'
```
Comment (sharadkjaiswal) [inline tools/install/DynamoInstaller.iss]: {#VersionInfoTextVersion} or higher is already installed. In order to proceed with the installation, you need to uninstall {#ProductName} {#Major}.{#Minor} manually.

**Comment by sharadkjaiswal** [comment]
LGTM

**Comment by nguyen-binh-minh** [comment]
@sharadkjaiswal updated as requested

**Pull request merged**